### ISSUE TITLE
[Bug] Multidimensional Multibinary action space support

### ISSUE THREAD

**Issue by ChengYen-Tang** [DLR-RM/stable-baselines3#1283]
### ❓ Question

We see N in MultiBinary, it may be a tuple.
https://github.com/openai/gym/blob/master/gym/spaces/multi_binary.py#L38-L44
If the above is the case, maybe there is an error here?
https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/preprocessing.py#L206-L208
https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/distributions.py#L682-L683

Looks like this issue?
https://github.com/DLR-RM/stable-baselines3/issues/1073

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by qgallouedec** [comment]
Small correction, the correct reference is in fact :

https://github.com/openai/gym/blob/c755d5c35a25ab118746e2ba885894ff66fb8c43/gym/spaces/multi_binary.py#L34

But I think you are right. If this is blocking you in your use case, I think using a wrapper to flatten the action space should work.

**Comment by ChengYen-Tang** [comment]
Will this issue be fixed in the future?

**Comment by qgallouedec** [comment]
As it is indeed a bug, yes. But as it concerns very few environments and a very simple hotfix exists, it is not a priority. However, if you want, contributions are welcome :)

**Comment by qgallouedec** [comment]
## To reproduce:

```python
from gym import spaces
from stable_baselines3.common.preprocessing import get_action_dim

get_action_dim(spaces.MultiBinary((3,2)))
```

```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/quentingallouedec/stable-baselines3/stable_baselines3/common/preprocessing.py", line 208, in get_action_dim
    return int(action_space.n)
TypeError: int() argument must be a string, a bytes-like object or a number, not 'tuple'
```

## System info

- OS: macOS-13.0.1-arm64-arm-64bit Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:30 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8103
- Python: 3.9.16
- Stable-Baselines3: 1.8.0a1
- PyTorch: 1.13.1
- GPU Enabled: False
- Numpy: 1.24.0
- Gym: 0.21.0

**Comment by ChengYen-Tang** [comment]
I have a question.
1. Is that right?
```
elif isinstance(action_space, spaces.MultiBinary):
    # Number of binary actions
    return int(action_space.n[0]) if type(action_space.n) in [tuple, list, np.ndarray] else int(action_space.n)
```

2. I don't know how to deal with it here.
Here may need to implement a MultiBernoulliDistribution?
https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/common/distributions.py#L682-L683

**Comment by araffin** [comment]
Hello,

> As it is indeed a bug, yes.

i would say it's more a feature request than a bug (and the feature is actually covered if you wrap the env). However,  I think we need to update the env checker.

**Comment by araffin** [comment]
Fixed in SB3 2.0 (proper error message is shown).
